specification/sql/resource-manager/Microsoft.Sql/2017-03-01-preview/managedDatabases.json

```diff
@@ -0,0 +1,556 @@
+          },
+          "Updates a managed database with maximal properties": {
+            "$ref": "./examples/ManagedDatabaseUpdateMax.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ManagedDatabaseProperties": {
+      "description": "The managed database's properties.",
+      "type": "object",
+      "properties": {
+        "collation": {
+          "description": "Collation of the managed database.",
+          "type": "string",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        },
+        "status": {
+          "description": "Status for the database.",
+          "enum": [
+            "Online",
+            "Offline",
+            "Shutdown",
+            "Creating",
+            "Inaccessible"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "ManagedDatabaseStatus",
+            "modelAsString": true
+          }
+        },
+        "creationDate": {
+          "format": "date-time",
+          "description": "Creation date of the database.",
+          "type": "string",
+          "readOnly": true
+        },
+        "pointInTime": {
+          "format": "date-time",
+          "description": "Conditional. If createMode is PointInTimeRestore, this value is required. Specifies the point in time (ISO8601 format) of the source database that will be restored to create the new database.",
+          "type": "string",
+          "readOnly": true
+        },
+        "defaultSecondaryLocation": {
+          "description": "Geo pared region.",
+          "type": "string",
+          "readOnly": true
+        },
+        "catalogCollation": {
+          "description": "Collation of the metadata catalog.",
+          "enum": [
+            "DATABASE_DEFAULT",
+            "SQL_Latin1_General_CP1_CI_AS"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "CatalogCollationType",
+            "modelAsString": true
+          },
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        },
+        "createMode": {
+          "description": "Managed database create mode. PointInTimeRestore: Create a database by restoring a point in time backup of an existing database. SourceDatabaseName, SourceManagedInstanceName and PointInTime must be specified. RestoreExternalBackup: Create a database by restoring from external backup files. Collation, StorageContainerUri and StorageContainerSasToken must be specified.",
+          "enum": [
+            "Default",
+            "RestoreExternalBackup",
+            "PointInTimeRestore"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "ManagedDatabaseCreateMode",
+            "modelAsString": true
+          },
+          "x-ms-mutability": [
+            "create"
+          ]
+        },
+        "sourceDatabaseName": {
```

Comment:
There are a bunch of conditional inputs. It might make sense to use the x-ms-discriminator to determine the shape of input based on the value of createMode. This will make the generated SDKs more usable. Details here - https://github.com/Azure/autorest/tree/master/docs/extensions/swagger-extensions-examples/x-ms-discriminator-value
@mcardosos @veronicagg  can provide guidance on this.

Reply:
I've done that based on postgresql swagger as an example and pushed the changes. Is there a way to generate SDKs based on my swagger on my machine before I push it here, so I can make sure it's good before I push?

Reply:
You can generate a SDK with [Autorest](https://github.com/Azure/autorest)